server/mcp/user_client.go

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) 2023-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package mcp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"maps"
+	"time"
+
+	"github.com/invopop/jsonschema"
+	"github.com/mark3labs/mcp-go/client"
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mattermost/mattermost-plugin-ai/server/llm"
+	"github.com/mattermost/mattermost/server/public/pluginapi"
+	orderedmap "github.com/wk8/go-ordered-map/v2"
+)
+
+const MMUserIDHeader = "X-Mattermost-UserID"
```

Comment:
This would allow someone to write an MCP server that trusted the Mattermost auth.